packages/contracts/contracts/L1/ValidatorManager.sol

```diff
@@ -0,0 +1,637 @@
+     */
+    function getCommissionMaxChangeRate(address validator) external view returns (uint8) {
+        return _validatorInfo[validator].commissionMaxChangeRate;
+    }
+
+    /**
+     * @notice Returns the number of started validators.
+     *
+     * @return The number of started validators.
+     */
+    function startedValidatorCount() external view returns (uint32) {
+        return _validatorTree.counter - _validatorTree.removed;
+    }
+
+    /**
+     * @notice Returns the weight of given validator. It not started, returns 0.
+     *         Note that `weight / startedValidatorTotalWeight()` is the probability that the
+     *         validator is selected as a priority validator.
+     *
+     * @param validator Address of the validator.
+     *
+     * @return The weight of given validator.
+     */
+    function getWeight(address validator) external view returns (uint120) {
+        return _validatorTree.nodes[_validatorTree.nodeMap[validator]].weight;
+    }
+
+    /**
+     * @notice Returns the jail expiration timestamp of given validator.
+     *
+     * @param validator Address of the jailed validator.
+     *
+     * @return The jail expiration timestamp of given validator.
+     */
+    function jailExpiresAt(address validator) external view returns (uint128) {
+        return _jail[validator];
+    }
+
+    /**
+     * @inheritdoc IValidatorManager
+     */
+    function updateValidatorTree(address validator, bool tryRemove) public {
+        if (tryRemove && getStatus(validator) == ValidatorStatus.STARTED) {
+            _validatorTree.remove(validator);
+        } else {
+            _validatorTree.update(validator, uint120(ASSET_MANAGER.reflectiveWeight(validator)));
+        }
+    }
+
+    /**
+     * @inheritdoc IValidatorManager
+     */
+    function nextValidator() public view returns (address) {
+        if (_nextPriorityValidator != address(0)) {
+            uint256 l2Timestamp = L2_ORACLE.nextOutputMinL2Timestamp();
+            if (block.timestamp >= l2Timestamp) {
+                uint256 elapsed = block.timestamp - l2Timestamp;
+                // If the current time exceeds one round time, it is a public round.
+                if (elapsed > ROUND_DURATION_SECONDS) {
+                    return Constants.VALIDATOR_PUBLIC_ROUND_ADDRESS;
+                }
+            }
+
+            return _nextPriorityValidator;
+        } else {
+            return TRUSTED_VALIDATOR;
+        }
+    }
+
+    /**
+     * @inheritdoc IValidatorManager
+     */
+    function getStatus(address validator) public view returns (ValidatorStatus) {
+        if (!_validatorInfo[validator].isInitiated) {
+            return ValidatorStatus.NONE;
+        }
+
+        if (_jail[validator] != 0) {
+            return ValidatorStatus.IN_JAIL;
+        }
+
+        if (ASSET_MANAGER.totalValidatorKro(validator) < MIN_REGISTER_AMOUNT) {
+            return ValidatorStatus.INACTIVE;
+        }
```

Comment:
Is this the case that the validator register successfully, then undelegate partially?
If so, can we unite the `ACTIVE` and `INACTIVE` to `PENDING`?
When the validator is `PENDING` status, validator should `start`. For starting validator, there are conditions:
- self-delegation is greater than `MIN_REGISTER_AMOUNT`
- delegation is greater than `MIN_START_AMOUNT`

In addition with this notation, `PENDING`, we can change the work `start` to `activate`. I thinkt `activate` is better than start. wdyt? In ethereum and other validator system, also activate is used.

Reply:
I think the separation between `ACTIVE` and `INACTIVE` is necessary because there are some actions an inactivate validator cannot do. (1. change commission rate, 2. receive new delegation)
I think to change `start` to `activate` is good idea, but I have no idea about the words to replace `INACTIVE` and `ACTIVE`. Do you have any suggestions?

Reply:
Ah, I see. How about these?
- INACTIVE -> PENDING
- ACTIVE -> READY
It's too hard to name it 😭

Reply:
Hmm READY is quite ambiguous because the validator cannot do anything but delegation in READY status. Also CAN_START has redundant meaning with READY. Let's keep thinking about it.. I have no idea 😢